dlib/pixel.h

```diff
@@ -72,6 +74,29 @@ namespace dlib
                 - min() == 0 
                 - max() == 255
                 - is_unsigned == true
+            - else if (bgr_layout == true) then
```

Comment:
This isn't a separate case from `rgb`.  The code sets both `rgb==true` and `bgr_layout==true` at the same time (which is what I think it should do).  But the docs should reflect that.  I also don't think there should be a `bgr_alpha` field.  A pixel with `pixel_traits::rgb_alpha == true && pixel_traits::bgr_layout == true` indicates it's a BGR layout pixel with an alpha for instance.  The docs should reflect this by updating the notes in the `rgb` and `rgb_alpha` `if` statement to indicate at such.